app/controllers/labels_controller.rb

```diff
@@ -2,12 +2,54 @@
 
 class LabelsController < ApplicationController
   load_and_authorize_resource
-  def search
-    results = Label.ransack(name_i_cont: params[:search]).result
-    paginated = results.paginate(per_page: 3, page: params[:page])
+
+  rescue_from CanCan::AccessDenied do |_exception|
+    redirect_to root_path, alert: t('controllers.authorization')
+  end
+
+  def index
+    @search = {}
+  end
+
+  def merge
+    label_ids = params[:label_ids]
+    new_name = params[:new_label_name]
+
+    if new_name.length.positive? && new_name.length < 15
+      new_label = Label.create(name: new_name)
+      new_label.tasks = Task.where(id: TaskLabel.select(:task_id).where(label_id: label_ids).distinct)
+      Label.where(id: label_ids).destroy_all
+    else
+      head :bad_request
```

Comment:
#merge now returns :unprocessable_entity on all errors. If you look into the function, you will see that I changed how the transaction rollback is handled again. After reading some threads on StackOverflow I think this is how you are supposed to handle a transaction rollback in rails.

Reply:
There is also RecordNotFound that can occur if non-existing label_ids are submitted. Unfortunately ActiveRecord::RecordNotFound has no .record.errors.full_messages so I would suggest to just use e.message.
With RecordNotFound there are now 4 errors that we catch. I think we could safely catch the parent ActiveRecord::ActiveRecordError instead.

Reply:
Yeah, I was checking the error classes as well and also identified that `RecordNotFound` doesn't provide `e.record` (which makes sense, since there is no record). I was also considering `ActiveRecordError`, but thought it would be too broad (also covering errors such as `ConnectionNotEstablished` or `StatementInvalid`), which we probably don't want to catch. Further, the `RecordNotFound` error should already be handled globally, simply returning a 404 (but of course, this can be validated again).

My other idea was to validate the potential changes (especially the update) or just add a second `rescue` block for other errors, such as the `RecordNotFound`.